### PR TITLE
Added restart, reset, poweroff, exit, reboot and shutdown commands.

### PR DESCRIPTION
**Pull request by ghost**
Added restart and reboot commands.
	/apps/terminal.asm Tested and working well.
	/docs/terminal.md  Updated.

### CODE REVIEW

**Comment by AtieP** [comment]
why did you need to rename `reset` to `reboot` and create an alias `restart`

**Comment by ghost** [comment]
Idk how to create alias

**Comment by ghost** [comment]
Added reset back

**Comment by ghost** [comment]
reset command back actually*

File: apps/terminal.asm [hunk 1]
```diff
@@ -147,6 +177,17 @@ main:
...
+.shutdown:
+    mov ax, 0x1000
+    mov ax, ss
+    mov sp, 0xf000
+    mov ax, 0x5307
+    mov bx, 0x0001
+    mov cx, 0x0003
+    int 0x15
+
+    ret  ;if interrupt doesnt work
```
Comment (AtieP) [inline apps/terminal.asm:189]: this thing here is wrong.

**Comment by ghost** [comment]
atie
added some good commands that nobody added

**Review by AtieP — CHANGES_REQUESTED** [review]
that ret doesn't belong there. also, the stack is modified and that can lead to chashes. and the restart command doesn't match with the label name.

File: apps/terminal.asm (hunk 1 above)
Comment (ghost) [reply apps/terminal.asm:189]: I tried it and it worked well

Comment (ghost) [reply apps/terminal.asm:189]: should I remove these lines?

186 187 188 and 189

**Comment by ghost** [comment]
Everything works well and restart command matches it's label `.reset`....

as I said it worked with me well...

**Comment by AtieP** [comment]
```x86asm
.COMMANDS.RESET: db "restart"
.COMMANDS.RESTART.LEN: equ $ - .COMMANDS.RESTART
.COMMANDS.RESTART: db "reset"
```
why.

```x86asm

.shutdown:
    mov ax, 0x1000
    mov ax, ss
    mov sp, 0xf000
    mov ax, 0x5307
    mov bx, 0x0001
    mov cx, 0x0003
    int 0x15

    ret  ;if interrupt doesnt work
```
first thing. the ret thing is going to crash the whole system. second thing, the stack is modified. if an interrupt is fired, bye bye stack. third thing. this shutdown doesn't work on all computers. i tested long ago in a 15 year old asus and it didn't work. also, the string doesn't match with the label name in some places.

**Comment by AtieP** [comment]
i think you should close this pr, and try again when you have more assembly knowledge. you also identified yourself as ahry before.

**Comment by ghost** [comment]
yeah I tried it in qemu

**Comment by AtieP** [comment]
>in qemu

if the shutdown is not guaranteed to work, then you should put an error message **and return using a jmp to the main loop rather than using a ret**.

**Pull request closed without merging**